kube-kvm/provision-kube-host-kvm.sh

```diff
@@ -0,0 +1,147 @@
+  rm -f "$RUNNING_VM_IMAGE"
+fi
+
+cp --reflink=auto "$KUBE_VM_IMAGE_NAME.qcow2" "$KUBE_VM_IMAGE_NAME-$KUBE_VM_NAME.qcow2"
+# TODO: Test if virbr0 is present
+virt-install \
+  --name "$KUBE_VM_NAME" \
+  --memory $(( KUBE_VM_MEM_GIB * 1024 )) \
+  --vcpus 1 \
+  --disk "$KUBE_VM_IMAGE_NAME-$KUBE_VM_NAME.qcow2" \
+  --import \
+  --network network=default \
+  --network bridge=br0 \
+  --os-variant opensuse42.2 \
+  --wait 0 \
+  --hvm
+
+echo "Sleeping 30 seconds to allow VM to start and obtain IP"
+sleep 30
+IP_WAIT_TIMEOUT=600
+unset KUBE_VM_DEFAULT_MAC
+while [[ -z "$KUBE_VM_DEFAULT_MAC" ]]; do
+  echo "Checking if default interface has MAC address"
+  KUBE_VM_DEFAULT_MAC=$(virsh domiflist "$KUBE_VM_NAME" | grep default | grep -oE '([[:xdigit:]]{2}:){5}[[:xdigit:]]{2}') || true
+  if [[ -z "$KUBE_VM_DEFAULT_MAC" ]]; then
+    if [[ $IP_WAIT_TIMEOUT -ge 0 ]]; then
+      (( IP_WAIT_TIMEOUT -= 15 ))
+      echo "MAC address not yet available. Retrying in 15 seconds"
+      sleep 15
+    else
+      echo "No MAC address for default interface on VM $KUBE_VM_NAME after 10 minutes. Check output of \`sudo virsh domiflist $KUBE_VM_NAME\`"
+      exit 1
+    fi
+  fi
+done
+unset KUBE_VM_DEFAULT_IP
+while [[ -z "$KUBE_VM_DEFAULT_IP" ]]; do
+  echo "Checking if IP lease exists for interface $KUBE_VM_DEFAULT_MAC"
+  KUBE_VM_DEFAULT_IP=$(virsh net-dhcp-leases default --mac "$KUBE_VM_DEFAULT_MAC" | grep -oE '([0-9]{1,3}\.){3}[0-9]{1,3}') || true
+  if [[ -z "$KUBE_VM_DEFAULT_IP" ]]; then
+    if [[ $IP_WAIT_TIMEOUT -ge 0 ]]; then
+      (( IP_WAIT_TIMEOUT -= 15 ))
+      echo "No IP assigned for $KUBE_VM_DEFAULT_MAC. Retrying in 15 seconds"
+      sleep 15
+    else
+      echo "No IP assigned for $KUBE_VM_DEFAULT_MAC after 10 minutes. Exiting."
+      exit 1
+    fi
+  fi
+done
+echo "Docker VM IP found: $KUBE_VM_DEFAULT_IP. Waiting for sshd to start"
+SSHD_TIMEOUT=60
+while [[ $SSHD_TIMEOUT -ge 0 ]]; do
+  nc $KUBE_VM_DEFAULT_IP 22 </dev/null | grep 'SSH-' && break
+  sleep 5
+  (( SSHD_TIMEOUT -= 5 ))
+done
+ssh-keygen -R $KUBE_VM_DEFAULT_IP
+ssh -o StrictHostKeyChecking=no -Ti "$KUBE_VM_KEY" vagrant@$KUBE_VM_DEFAULT_IP -- sudo bash -o errexit << EOF
+hostname $KUBE_VM_NAME
+echo $KUBE_VM_NAME > /etc/hostname
+cp /etc/sysconfig/network/ifcfg-eth0 /etc/sysconfig/network/ifcfg-eth1
+echo DHCLIENT_SET_DEFAULT_ROUTE='yes' /etc/sysconfig/network/ifcfg-eth1
+wicked ifup eth1
+EOF
+
+KUBE_VM_BRIDGED_IP=$(ssh -i "$KUBE_VM_KEY" vagrant@$KUBE_VM_DEFAULT_IP 'ip -4 addr show eth1' | tr / ' ' | awk '/inet/ { print $2 }')
+if [[ -z "$KUBE_VM_BRIDGED_IP" ]]; then
+  echo "There was a problem getting the IP of the bridged interface. See output of \`ifconfig eth1\` in VM $KUBE_VM_NAME"
+  exit 1
+fi
+
+ssh-keygen -R $KUBE_VM_BRIDGED_IP
+ssh -o StrictHostKeyChecking=no -Ti "$KUBE_VM_KEY" vagrant@$KUBE_VM_BRIDGED_IP  << EOF
+git clone https://github.com/suse/scf
+cd scf
+sudo bash -o errexit << EOSU
+export SCF_BIN_DIR=/usr/local/bin
+./bin/common/install_tools.sh
```

Comment:
Why does the VM need tools again? I thought it was meant to be a kube-only VM (and not even for actually running `kubectl` on; rather, just running the kube server part)

Reply:
The image itself is for a kube-only VM.. installing common tools installs helm, brings in a newer version of kubectl (IIRC), and also adds 'k' and 'cf'. The intention of this script is to provide a base to deploy SCF to, with the assumption that things like bosh release fetching and compilation, and fissile image creation, don't need to happen on it.

Reply:
Right, but the _server_ has no need of `kubectl`, `k`, nor `cf`. `helm` is only useful insofar as installing `tiller` (which can be done on the machine the script is running from).

Running `install_tools.sh` on the server means cloning all of SCF 😞

Reply:
True, it doesn't *need* `kubectl`, `k`, and `cf`, but I'd argue they will be useful when debugging when things go wrong in CI (kind of like how the SCF images don't need sshd, less, vi, sed, grep), and reusing `install_tools.sh` ensures we have the same helm version across deployments

Cloning SCF without submodules is fairly quick (it's about 38 MB). I could also cherry-pick the versions.sh and `common/install_tools.sh` to avoid the need to clone it at all.

Reply:
I'm confused. You don't expect people to clone scf to get the shared version, but expect cloning it to be quick here?

I'd just go with, let the user do it if they want to, it's already in the readme.

Reply:
Users don't need to clone scf to get the version if the version is kept current :) The common tool installation isn't in the README (yet); I can add it there, and remove it from default installation.. I still think this will result in more time spent picking in these tools manually, installing helm manually, and typing out full kubectl commands because 'k' isn't around

Reply:
Right, the problem is that I totally forgot this is the cf-ci repo, not the scf repo. So it makes sense.